### DISCUSSION TITLE
Install-Relaunch - restore view?

### DISCUSSION THREAD

**Discussion by meteorquake** [sumatrapdfreader/sumatrapdf#2897] [Ideas]
Sumatra regularly has updates which it offers when you start up, which is good, but when you click so it automatically installs and relaunches, it doesn't restore the file/view you had opened, so typically I have to relaunch the file, which may be from Explorer though of course if it had been launched from some process it may not be advisable or be slow to rerun the task that had made the launch. This is rather an annoyance for the simple reason I suspect it could easily be made to restore the view :)
David

**Comment by tactilis** [comment]
> it doesn't restore the file/view you had opened,

Do you have this option set?

![image](https://user-images.githubusercontent.com/4060092/186646643-7423db02-3c06-4756-acbb-9c3e8be0236e.png)

**Comment by meteorquake** [comment]
I don't because obviously I would want files to reopen when a restart (from an update) is enacted, but not after an ordinary launch

**Comment by tactilis** [comment]
@meteorquake
Since you don't want SumatraPDF to remember and re-open files when you are using it normally for document reading, then why not do this:

1. Note on start up that SumatraPDF is offering you an update.
2. Decline the update.
3. Read your document, and then when you have finished...
4. Do a Check for Updates and accept the update.

![image](https://user-images.githubusercontent.com/4060092/186676275-94e6cc93-5661-4149-8ac2-04c0ce89d608.png)

The restart after the update won't reopen the current document, but that fits your stated mode of usage.

**Comment by meteorquake** [comment]
That's more clicking than having to relaunch the document :) Usually if there's an update I'd want it immediately rather than deferred.
I was hoping it would be a minor improvement easy to be done, as I imagine anyone doing an update-restart would want files reopened, whatever their normal setting.
d

**Reply by GitHubRulesOK** [reply]
Hmm
 I was one update behind so with two files open and zoomed in on tab one I ran update to(yester)-days version, no problem it uninstalled the search and preview dlls, then replaced those and relaunched (on demand) with the same zoomed in view of first of two tabs
